src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -488,12 +488,47 @@ public bool SpansPreprocessorDirective(IEnumerable<SyntaxToken> tokens)
         private bool SpansPreprocessorDirective(SyntaxTriviaList list)
             => list.Any(t => IsPreprocessorDirective(t));
 
-        public bool IsInHeader(SyntaxNode node, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
+        public bool IsOnHeader(int position, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader)
         {
             var start = GetStartOfNodeExcludingAttributes(ownerOfHeader);
             var end = lastTokenOrNodeOfHeader.FullSpan.End;
 
-            return node.Span.Start >= start && node.Span.End <= end;
+            return start <= position && position <= end;
+        }
+
+        public bool IsInHeader(int position, SyntaxNode ownerOfHeader, SyntaxNodeOrToken lastTokenOrNodeOfHeader, ImmutableArray<SyntaxNode> holes)
```

Comment:
`public bool IsInheader<THoleSyntax>(..., ImmutableArray<THoleSyntax> holes) where THoleSyntax : SyntaxNode`

Now you can pass `ImmutableArray<WhateverSyntaxType>` here and not have to precast to SyntaxNode.